File: Python/project/app/modules/database_module/models/default/company.py
from tortoise import fields

from app.modules.database_module.models.database_model import DatabaseModel
from app.modules.database_module.models.default.offer import Offer
from app.modules.database_module.models.default.work_tutor import WorkTutor


class Company(DatabaseModel):
    name = fields.CharField(max_length=100, unique=True)
    sector = fields.TextField()
    web = fields.TextField(null=True)
    cif = fields.CharField(max_length=9, unique=True)
    logo = fields.TextField(null=True)
    security_code = fields.CharField(max_length=5, unique=True)

    # Relations
    work_tutors: fields.ReverseRelation["WorkTutor"]
    offers: fields.ReverseRelation["Offer"]

File: iOS/Modules/Sources/Domain/Model/Company/Company.swift
import Foundation

public struct Company: Identifiable, Sendable, Codable {
    public let id: Int
    public let name: String
    public let sector: String
    public let web: String?
    public let cif: String
    public let logo: String?
    public let securityCode: String?

    enum CodingKeys: String, CodingKey {
        case id, name, sector, web, cif, logo
        case securityCode = "security_code"
        }
    public init(
        id: Int,
        name: String,
        sector: String,
        web: String?,
        cif: String,
        logo: String?,
        securityCode: String?
    ) {
        self.id = id
        self.name = name
        self.sector = sector
        self.web = web
        self.cif = cif
        self.logo = logo
        self.securityCode = securityCode
    }
}
